coupled_laplace_problem/coupled_laplace_problem.cc

```diff
@@ -0,0 +1,652 @@
+  output_results() const;
+
+  Triangulation<dim> triangulation;
+  FE_Q<dim>          fe;
+  DoFHandler<dim>    dof_handler;
+  MappingQ1<dim>     mapping;
+
+  SparsityPattern      sparsity_pattern;
+  SparseMatrix<double> system_matrix;
+
+  Vector<double> solution;
+  Vector<double> old_solution;
+  Vector<double> system_rhs;
+
+  // We allocate all structures required for the preCICE coupling. The map
+  // is used to apply Dirichlet boundary conditions and filled in the Adapter
+  // class with data from the other participant. The CouplingParameters hold the
+  // preCICE configuration as described above. The interface boundary ID is the
+  // ID associated to our coupling interface and needs to be specified, when we
+  // set up the Adapter class object, because we pass it directly to the
+  // Constructor of this class.
+  std::map<types::global_dof_index, double> boundary_data;
+  CouplingParamters                         parameters;
+  const types::boundary_id                  interface_boundary_id;
+  Adapter<dim, CouplingParamters>           adapter;
+
+  // The time-step size delta_t is the acutual time-step size used for all
+  // computations. The preCICE time-step size is obtained by preCICE in order to
+  // ensure a synchronization at all coupling time steps. The solver time
+  // step-size is the desired time-step size of our individual solver. In more
+  // sophisticated computations, it might be determined adaptively. The
+  // `time_step` counter is just used for the time-step number.
+  double       delta_t;
+  double       precice_delta_t;
+  const double solver_delta_t = 0.1;
+  unsigned int time_step      = 0;
+};
+
+
+
+template <int dim>
+class RightHandSide : public Function<dim>
+{
+public:
+  virtual double
+  value(const Point<dim> &p, const unsigned int component = 0) const override;
+};
+
+
+
+template <int dim>
+class BoundaryValues : public Function<dim>
+{
+public:
+  virtual double
+  value(const Point<dim> &p, const unsigned int component = 0) const override;
+};
+
+template <int dim>
+double
+RightHandSide<dim>::value(const Point<dim> &p,
+                          const unsigned int /*component*/) const
+{
+  double return_value = 0.0;
+  for (unsigned int i = 0; i < dim; ++i)
+    return_value += 4.0 * std::pow(p(i), 4.0);
+
+  return return_value;
+}
+
+
+template <int dim>
+double
+BoundaryValues<dim>::value(const Point<dim> &p,
+                           const unsigned int /*component*/) const
+{
+  return p.square();
+}
+
+
+
+template <int dim>
+CoupledLaplaceProblem<dim>::CoupledLaplaceProblem()
+  : fe(1)
+  , dof_handler(triangulation)
+  , interface_boundary_id(1)
+  , adapter(parameters, interface_boundary_id)
+{}
+
+
+template <int dim>
+void
+CoupledLaplaceProblem<dim>::make_grid()
+{
+  GridGenerator::hyper_cube(triangulation, -1, 1);
+  triangulation.refine_global(4);
+
+  for (const auto &cell : triangulation.active_cell_iterators())
+    for (auto f : GeometryInfo<dim>::face_indices())
```

Comment:
```suggestion
    for (const auto face : cell->face_iterators())
```
Consider `GeometryInfo` as deprecated ;)